exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -135,6 +138,7 @@
   private final ForemanResult foremanResult = new ForemanResult();
   private final ConnectionClosedListener closeListener = new ConnectionClosedListener();
   private final ChannelFuture closeFuture;
+  private QueryWarning queryWarning;
```

Comment:
Please move this, and the following methods to the QueryManager where other state is maintained.